packages/multichain-account-service/src/MultichainAccountService.ts

```diff
@@ -89,6 +103,72 @@ export class MultichainAccountService {
       },
       selectKeyringControllerKeyrings,
     );
+
+    // Reverse mapping between account ID and their multichain wallet/account:
+    // QUESTION: Should we move the reverse mapping logic to the
+    // `MultichainAccount{,Wallet}` implementation instead? For now they do not
+    // store any accounts and they heavily rely on the `AccountProvider`s, which
+    // makes it hard to implement it there...
+    for (const wallet of this.#wallets.values()) {
+      for (const multichainAccount of wallet.getMultichainAccounts()) {
+        for (const account of multichainAccount.getAccounts()) {
+          this.#reverse.set(account.id, {
+            wallet,
+            multichainAccount,
+          });
+        }
+      }
+    }
+
+    this.#messenger.subscribe('AccountsController:accountAdded', (account) =>
+      this.#handleOnAccountAdded(account),
+    );
+    this.#messenger.subscribe('AccountsController:accountRemoved', (id) =>
+      this.#handleOnAccountRemoved(id),
+    );
+  }
+
+  #handleOnAccountAdded(account: InternalAccount): void {
+    // We completely omit non-BIP-44 accounts!
+    if (!isBip44Account(account)) {
+      return;
+    }
+
+    const wallet = this.#wallets.get(
+      toMultichainAccountWalletId(account.options.entropy.id),
+    );
+    if (wallet) {
+      // This new account might be a new multichain account, and the wallet might not
+      // know it yet, so we need to force-sync here.
+      wallet.sync();
+
+      // We should always have a wallet here, since we are refreshing that
+      // list when any keyring's states got changed.
+      const multichainAccount = wallet.getMultichainAccount(
+        account.options.entropy.groupIndex,
+      );
+      if (multichainAccount) {
+        // Same here, this account should have been already grouped in that
+        // multichain account.
+        this.#reverse.set(account.id, {
+          wallet,
+          multichainAccount,
+        });
+      }
+    }
+  }
+
+  #handleOnAccountRemoved(id: InternalAccount['id']): void {
+    // Force sync of the appropriate wallet if an account got removed.
+    const found = this.#reverse.get(id);
+    if (found) {
+      const { wallet } = found;
+
+      wallet.sync();
```

Comment:
Same sentiment as above re: adding an account.

Reply:
I did refactor this, lemme know what you think. (I still believe we could do it in a more surgical way, but that should be ok for now).